float32[12] data